### DISCUSSION TITLE
Why peers cannot discover each other?

### DISCUSSION THREAD

**Discussion by ghost** [libp2p/rust-libp2p#4056] [Q&A] [answered]
Code here: https://github.com/SightNet/sightnet-app/blob/master/src/app.rs
Logs:
```sh
Local node is listening on /ip4/127.0.0.1/udp/48367/quic-v1
Local node is listening on /ip4/xxx.xxx.xxx/udp/48367/quic-v1
Local node is listening on /ip4/xxx.xxx.xxx/udp/48367/quic-v1
Local node is listening on /ip4/127.0.0.1/tcp/36085
Local node is listening on /ip4/xxx.xxx.xxx/tcp/36085
Local node is listening on /ip4/xxx.xxx.xxx/tcp/36085
```
But if i start two instances, they won't discover each other(so, i can send messages: Publish error: InsufficientPeers)
Later, it worked.
I tried also official example: https://github.com/libp2p/rust-libp2p/blob/master/examples/chat-example/.
Also i tried to change versions of library in example.
And there is the same problem.
If it is important, i use vpn connection.

**Comment by thomaseizinger** [comment]
Are your peers on the same network? The chat example uses mDNS for discovery which will only work on the local network.

**Reply by ghost** [reply]
Yes, on the same network.

**Reply by ghost** [reply]
![image](https://github.com/libp2p/rust-libp2p/assets/129421732/07894913-4afe-4893-b47a-d629f93fe344)

**Reply by thomaseizinger** [reply]
Try logging the additional `SwarmEvent` variant and see if there is something useful in there.

I don't see anything wrong with the code at first sight.

**Reply by ghost** [reply]
> Try logging the additional `SwarmEvent` variant and see if there is something useful in there.
> 
> I don't see anything wrong with the code at first sight.

I dont think, that there is problem in my code. On the screen i run two instances of official [example](https://github.com/libp2p/rust-libp2p/blob/master/examples/chat-example/).
And i tried to add this code: 
![image](https://github.com/libp2p/rust-libp2p/assets/129421732/71e5fb18-5625-40e3-9914-42b202661d97)
And there are no (other) events.

**Reply by ghost** [reply]
So, i have just built and run sightnet-app on my mobile(via termux) and it works fine.

**Reply by thomaseizinger** [reply]
What OS are you running on your desktop?

**Reply by ghost** [reply]
> What OS are you running on your desktop?

EndeavourOS

**Reply by thomaseizinger** [reply, marked as answer]
Can you somehow verify whether mDNS works on your machine at all?

It relies on multicast so if that is disabled for example then things won't work.

**Reply by ghost** [reply]
I am stupid. Just installed [avahi](https://wiki.archlinux.org/title/Avahi) and disabled firewall and problem disappeared... Thank you.
